integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java

```diff
@@ -0,0 +1,192 @@
+import java.util.Map;
+
+import oracle.kubernetes.operator.utils.Domain;
+import oracle.kubernetes.operator.utils.ExecCommand;
+import oracle.kubernetes.operator.utils.ExecResult;
+import oracle.kubernetes.operator.utils.Operator;
+import oracle.kubernetes.operator.utils.TestUtils;
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+/**
+ * Simple JUnit test file used for testing Operator.
+ *
+ * <p>This test is used for creating Operator(s) and domain(s) which are managed by the Operator(s).
+ * WebLogic docker image is created by WebLogic Image Tool
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ItImageTool extends BaseTest {
+  private static final String TEST_RESOURCE_LOC = "integration-tests/src/test/resources";
+  private static String TEST_APP_PROPS_FILE = "OperatorWIT.properties";
+  private static String WLS_IMAGE_VERSION = "12.2.1.3.0";
+  private static String WLS_IMAGE_NAME = "imagetool/build/weblogic";
+  private static String WLS_IMAGE_TAG;
+
+  private static Operator operator;
+  private static Domain domain;
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It creates
+   * a WebLogic docker image using WebLogic Image Tool. It does the initialization of the integration
+   * test properties defined in OperatorIT.properties and setting the resultRoot, pvRoot and projectRoot attributes.
+   * It also creates Operator, domain and a test domain yaml file.
+   *
+   * @throws Exception exception
+   */
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    if (FULLTEST) {
+      // Determine image name and version to be used
+      if (System.getenv("IMAGE_NAME_WEBLOGIC") != null) {
+        WLS_IMAGE_NAME = System.getenv("IMAGE_NAME_WEBLOGIC");
+      }
+
+      if (System.getenv("IMAGE_TAG_WEBLOGIC") != null) {
+        WLS_IMAGE_VERSION = System.getenv("IMAGE_TAG_WEBLOGIC");
+      }
+
+      WLS_IMAGE_TAG = WLS_IMAGE_NAME + ":" + WLS_IMAGE_VERSION;
+      logger.info("WebLogic image name is: " + WLS_IMAGE_TAG);
+
+      // env vars IMAGE_NAME_WEBLOGIC and IMAGE_TAG_WEBLOGIC overwrite the wls docker image
+      // specified in OperatorIT.properties
+      if (System.getenv("IMAGE_NAME_WEBLOGIC") != null && System.getenv("IMAGE_TAG_WEBLOGIC") != null) {
+        TEST_APP_PROPS_FILE = APP_PROPS_FILE;
+      }
+      logger.info("Using <" + TEST_APP_PROPS_FILE + "> to create Operator and Domain");
+
+      // Build WebLogic docker image using imagetool
+      buildWlsDockerImage();
+
+      // initialize test properties and create the directories
+      initialize(TEST_APP_PROPS_FILE);
+
+      // Create operator1
+      if (operator == null) {
+        logger.info("Creating Operator & waiting for the script to complete execution");
+        operator = TestUtils.createOperator(OPERATOR1_YAML);
```

Comment:
Is it possible to create the operator and Docker image in parallel?

Reply:
Vanaja started a Parallel run proj. not sure whether it does it or not. But it's a good test approach for vanaja to consider?